services/api/service.go

```diff
@@ -1069,45 +1089,60 @@ func (api *RelayAPI) handleRegisterValidator(w http.ResponseWriter, req *http.Re
 		// Ensure a valid timestamp (not too early, and not too far in the future)
 		registrationTimestamp := signedValidatorRegistration.Message.Timestamp.Unix()
 		if registrationTimestamp < int64(api.genesisInfo.Data.GenesisTime) { //nolint:gosec
-			handleError(regLog, http.StatusBadRequest, "timestamp too early")
-			return
+			logAndReturnError(regLog, http.StatusBadRequest, "timestamp too early", nil)
+			break
 		} else if registrationTimestamp > registrationTimestampUpperBound {
-			handleError(regLog, http.StatusBadRequest, "timestamp too far in the future")
-			return
+			logAndReturnError(regLog, http.StatusBadRequest, "timestamp too far in the future", nil)
+			break
 		}
 
 		// Check if a real validator
 		isKnownValidator := api.datastore.IsKnownValidator(pkHex)
 		if !isKnownValidator {
-			handleError(regLog, http.StatusBadRequest, fmt.Sprintf("not a known validator: %s", pkHex))
-			return
+			logAndReturnError(regLog, http.StatusBadRequest, fmt.Sprintf("not a known validator: %s", pkHex), nil)
+			break
 		}
 
-		// Check for a previous registration timestamp
-		prevTimestamp, err := api.redis.GetValidatorRegistrationTimestamp(pkHex)
+		// Check for a previous registration timestamp and see if fields changed
+		cachedRegistrationData, err := api.redis.GetValidatorRegistrationData(pkHex)
+		haveCachedRegistration := cachedRegistrationData != nil
+
 		if err != nil {
-			regLog.WithError(err).Error("error getting last registration timestamp")
-		} else if prevTimestamp >= uint64(signedValidatorRegistration.Message.Timestamp.Unix()) { //nolint:gosec
-			// abort if the current registration timestamp is older or equal to the last known one
-			return
+			regLog.WithError(err).Error("error getting last registration timestamp") // continue to validation + processing
```

Comment:
I don't believe this should be an error. This will happen for any new validator. I think info is more appropriate. Also the message is a little outdated now, just remove "timestamp".

Reply:
added a comment to clarify